File: week6/c++/week6_8.cpp
#include <iostream>
#include <algorithm>
#include <vector>

using namespace std;

int main(int argc, char **argv)
{
    int test_case;
    int T;
    cin >> T;

    for (test_case = 1; test_case <= T; ++test_case)
    {
        int n;
        cin >> n;
        vector<int> a(n);
        for (int i = 0; i < n; i++)
        {
            cin >> a[i];
        }

        sort(a.begin(), a.end());
        int answer = 0;
        for (int i = 0; i < n; i++)
        {
            answer += a[i];
        }
        answer = answer + n + a[n - 1];

        cout << "#" << test_case << " ";
        cout << answer << endl;
    }
    return 0;
}

File: week6/java/week6_8.java
import java.lang.reflect.Array;
import java.util.Arrays;
import java.util.Scanner;

public class week6_8 {
    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);

        int t = sc.nextInt();

        for(int test_case=1; test_case<=t; test_case++){
            int n = sc.nextInt();
            int[] arr = new int[n];

            for(int i=0; i<n; i++){
                arr[i] = sc.nextInt();
            }

            Arrays.sort(arr);

            int answer = 0;

            for(int i=0; i<arr.length; i++){
                answer+=2*arr[i]+1;
                if(i>0){
                    answer -= arr[i-1];
                }
            }

            System.out.println("#"+test_case+" "+answer);
        }
    }
}
